bin/livy-server

```diff
@@ -18,34 +18,150 @@
 #
 # Runs Livy server.
 
-set -e
+
+usage="Usage: livy-server (start|stop)"
+
+# Find the java binary
+if [ -n "${JAVA_HOME}" ]; then
+  RUNNER="${JAVA_HOME}/bin/java"
+else
+  if [ `command -v java` ]; then
+    RUNNER="java"
+  else
+    echo "JAVA_HOME is not set" >&2
+    exit 1
+  fi
+fi
 
 export LIVY_HOME=$(cd $(dirname $0)/.. && pwd)
-cd $LIVY_HOME
+export LIVY_CONF_DIR="${LIVY_CONF_DIR:-"${LIVY_HOME}/conf"}"
```

Comment:
"export" is probably not needed here. Also this change now conflicts with a change I pushed recently, so it needs to be updated.